core/arch/x86/x86.asm

```diff
@@ -649,6 +652,9 @@ cat_have_lock:
         mov      REG_XSI, [2*ARG_SZ + REG_XBP]  /* sysnum */
         pop      REG_XAX             /* syscall */
         pop      REG_XCX             /* dstack */
+#if defined(LINUX) && !defined(X64)
+        lea      REG_XSP, [ARG_SZ + REG_XSP] /* undo align-16 lea from above */
+#endif
```

Comment:
I wonder if you could add an exit event that calls check_stack_alignment() to suite/tests/client-interface/cleancall.dll.c and confirm it crashes without this fix? Then we'll have a regression test.

Reply:
How can I run this specific test?

Reply:
`ctest -V -R 'client\.cleancall$'`